x-pack/plugins/aiops/common/api/stream_reducer.ts

```diff
@@ -45,6 +47,8 @@ export function streamReducer(
         return cp;
       });
       return { ...state, changePoints };
+    case API_ACTION_NAME.ERROR:
+      return { ...state, errors: [...state.errors, action.payload] };
     case API_ACTION_NAME.RESET:
       return initialState;
     case API_ACTION_NAME.UPDATE_LOADING_STATE:
```

Comment:
do you need to reset errors on `UPDATE_LOADING_STATE`?

Reply:
The attributes updated related to "loading state" do not cover errors and we want to retain errors while the analysis might still continue. Only on a reset (e.g. when starting a new request) the original default state without errors will be restored.